pkg/acioperator/operator.go

```diff
@@ -0,0 +1,371 @@
+	log.Println(cmd)
+	stdout, err := cmd.Output()
+
+	log.Info(string(stdout))
+
+	k8sclient := getk8sClient()
+
+
+	deploymentsClient := k8sclient.AppsV1().Deployments("aci-containers-system")
+	t.Deployment, _ = deploymentsClient.Get("aci-containers-controller",metav1.GetOptions{})
+	if t.Deployment == nil {
+		log.Info("aci-containers-controller deployment is nil..returning")
+		return
+	}
+
+	t.Deployment.OwnerReferences = []metav1.OwnerReference{
+					*metav1.NewControllerRef(acicontainersoperator, operators.SchemeGroupVersion.WithKind("AciContainersOperator")),
+				}
+
+	_, err = deploymentsClient.Update(t.Deployment)
+	if err != nil {
+		fmt.Println(err.Error())
+	}
+
+
+	hostdaemonsetclient := k8sclient.AppsV1().DaemonSets("aci-containers-system")
+	t.HostDaemonset, _ = hostdaemonsetclient.Get("aci-containers-host",metav1.GetOptions{})
+	if t.HostDaemonset == nil {
+		log.Info("aci-containers-host daemonset is nil.....returning")
+		return
+	}
+
+	t.HostDaemonset.OwnerReferences = []metav1.OwnerReference{
+		*metav1.NewControllerRef(acicontainersoperator, operators.SchemeGroupVersion.WithKind("AciContainersOperator")),
+	}
+
+	_, err = hostdaemonsetclient.Update(t.HostDaemonset)
+	if err != nil {
+		fmt.Println(err.Error())
+	}
+
+	ovsdaemonsetclient := k8sclient.AppsV1().DaemonSets("aci-containers-system")
+	t.OvsDaemonset, _ = hostdaemonsetclient.Get("aci-containers-openvswitch",metav1.GetOptions{})
+	if t.OvsDaemonset == nil {
+		log.Info("aci-containers-openvswitch daemonset is nil.....returning")
+		return
+	}
+
+	t.OvsDaemonset.OwnerReferences = []metav1.OwnerReference{
+		*metav1.NewControllerRef(acicontainersoperator, operators.SchemeGroupVersion.WithKind("AciContainersOperator")),
+	}
+
+	_, err = ovsdaemonsetclient.Update(t.OvsDaemonset)
+	if err != nil {
+		fmt.Println(err.Error())
+	}
+
+	log.Info("Platform flavor is ",acicontainersoperator.Spec.Flavor)
+
+	if (acicontainersoperator.Spec.Flavor == "OpenShift") {
+
+		log.Info("Updating Network Status of the Openshift Object....")
+
+		cfg, err := config.GetConfig()
+		scheme := runtime.NewScheme()
+
+		rclient, err := client.New(cfg, client.Options{Scheme: scheme})
+		if err != nil {
+			return
+		}
+
+		clusterConfig := &configv1.Network{
+			TypeMeta:   metav1.TypeMeta{APIVersion: configv1.GroupVersion.String(), Kind: "Network"},
+			ObjectMeta: metav1.ObjectMeta{Name: "cluster"},
+		}
+
+		err = rclient.Get(context.TODO(), types.NamespacedName{
+			Name: "cluster",
+		}, clusterConfig)
+		if err != nil {
+			log.Info(err)
+		}
+		log.Info("Config Spec is  ", clusterConfig.Spec)
+		clusterConfig.Status.ClusterNetwork = clusterConfig.Spec.ClusterNetwork
+		clusterConfig.Status.NetworkType = clusterConfig.Spec.NetworkType
+		clusterConfig.Status.ServiceNetwork = clusterConfig.Spec.ServiceNetwork
+
+		log.Info("Status is ", clusterConfig.Status)
+
+		ctx := context.TODO()
+		err = rclient.Update(ctx, clusterConfig)
```

Comment:
if your update fails you need to requeue the request here. there is a chance that etcd writes can fail here